packages/core/types/Fields.ts

```diff
@@ -90,7 +90,7 @@ export type ExternalField<
     query: string;
     filters: Record<string, any>;
   }) => Promise<any[] | null>;
-  mapProp?: (value: any) => Props;
+  mapProp?: (value: any,filters:any) => Props;
```

Comment:
The second param would probably be better as an object, and we may as well use the correct type for `filters` -

```suggestion
  mapProp?: (value: any, options: { filters: Record<string, any> }) => Props;
```